apps/opik-frontend/src/components/pages/PlaygroundPage/PlaygroundOutputs/PlaygroundOutputScores/PlaygroundOutputScores.tsx

```diff
@@ -0,0 +1,356 @@
+    }
+    return names;
+  }, [selectedRules]);
+
+  // Track if we're waiting for specific rules to load
+  // This prevents premature polling stop when rules haven't loaded yet
+  // Note: if rulesLoaded is true but selectedRules is empty, the rules were deleted
+  const hasSpecificRulesSelected =
+    Array.isArray(selectedRuleIds) && selectedRuleIds.length > 0;
+  const rulesStillLoading =
+    hasSpecificRulesSelected && !rulesLoaded && selectedRules.length === 0;
+
+  // Check if selected rules were deleted (specific rules selected, loaded, but none found)
+  const selectedRulesDeleted =
+    hasSpecificRulesSelected && rulesLoaded && selectedRules.length === 0;
+
+  // Store values in refs for access in refetchInterval
+  const expectedScoreNamesRef = useRef<Set<string>>(expectedScoreNames);
+  expectedScoreNamesRef.current = expectedScoreNames;
+  const rulesStillLoadingRef = useRef(rulesStillLoading);
+  rulesStillLoadingRef.current = rulesStillLoading;
+
+  // Fetch trace to get scores
+  // Disable query if selected rules were deleted (nothing to poll for)
+  const { data: trace } = useTraceById(
+    { traceId: traceId! },
+    {
+      enabled: !!traceId && hasSelectedRules && !selectedRulesDeleted,
+      refetchInterval: (query) => {
+        // Use ref to track elapsed time since polling started
+        const startTime = pollingStartTimeRef.current || Date.now();
+        const elapsedTime = Date.now() - startTime;
+
+        // Always stop if we've exceeded max polling time
+        if (elapsedTime > MAX_REFETCH_TIME) {
+          return false;
+        }
+
+        // If rules are still loading, keep polling
+        if (rulesStillLoadingRef.current) {
+          return REFETCH_INTERVAL;
+        }
+
+        const data = query.state.data;
+        const receivedScores = data?.feedback_scores ?? [];
+        const expectedNames = expectedScoreNamesRef.current;
+
+        // If specific rules were selected, wait for all expected scores
+        if (expectedNames.size > 0) {
+          const receivedNames = new Set(receivedScores.map((s) => s.name));
+          const allExpectedReceived = [...expectedNames].every((name) =>
+            receivedNames.has(name),
+          );
+          if (allExpectedReceived) {
+            return false;
+          }
+        } else {
+          // For "all" selection or legacy data, stop as soon as we have any scores
+          if (receivedScores.length > 0) {
+            return false;
+          }
+        }
+
+        return REFETCH_INTERVAL;
+      },
+    },
+  );
+
+  const feedbackScores: TraceFeedbackScore[] = useMemo(() => {
+    return trace?.feedback_scores ?? [];
+  }, [trace?.feedback_scores]);
+
+  // Create a map from score name to score for quick lookup
+  const scoresByName = useMemo(() => {
+    const map: Record<string, TraceFeedbackScore> = {};
+    for (const score of feedbackScores) {
+      map[score.name] = score;
+    }
+    return map;
+  }, [feedbackScores]);
```

Comment:
```suggestion
const scoresByName = useMemo(() => 
  Object.fromEntries(feedbackScores.map((s) => [s.name, s])), 
[feedbackScores]);
```

It's also possible with keyBy

```
import keyBy from "lodash/keyBy";

const scoresByName = useMemo(() => 
  keyBy(feedbackScores, 'name'), 
[feedbackScores]);
```

Both are good, maybe lodash is cleaner & better for maintenance (The string is a bit anti typescript, but it's better in maintenance compared to **map, array, & fromEntries** syntax )